src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -889,6 +897,73 @@ private JButton getCleanTableButton() {
 		return resetButton;
 	}
 
+	private JButton getValidateButton() {
+		validateButton = new JButton(MessageBundle.getMessage("angal.inventory.validate.btn"));
+		validateButton.setMnemonic(MessageBundle.getMnemonic("angal.inventory.validate.btn.key"));
+		if (inventory == null) {
+			validateButton.setEnabled(false);
+		}
+		validateButton.addActionListener(actionEvent -> {
+				if (inventory == null) {
+					MessageDialog.info(null, "angal.inventory.inventorymustsavebeforevalidate.msg");
+					return;
+				}
+				List<MedicalInventoryRow> invRowWithoutLot = inventoryRowSearchList.stream().filter(invRow -> invRow.getLot() == null).collect(Collectors.toList());
+				if (invRowWithoutLot.size() > 0) {
+					MessageDialog.info(null, "angal.inventory.allinventoryrowshouldhavelotbeforevalidate.msg");
+					return;
+				}
+				int reset = MessageDialog.yesNo(null, "angal.inventoryrow.doyoureallywanttovalidatethisinventory.msg");
+				if (reset == JOptionPane.YES_OPTION) {
+					String dischargeCode = inventory.getDischargeType();
+					String chargeCode = inventory.getChargeType();
+					Integer supplierId = inventory.getSupplier();
+					String wardCode = inventory.getDestination();
+					if (dischargeCode == null || dischargeCode.equals("")) {
+						MessageDialog.info(null, "angal.inventory.choosedischargetypebeforevalidate.msg");
```

Comment:
Are all these dialogs `info`?  They seem to read to me as `error`

Reply:
okay fixed